code/modules/antagonists/wizard/equipment/artefact.dm

```diff
@@ -169,7 +169,12 @@
 	if(M.stat != DEAD)
 		to_chat(user, "<span class='warning'>This artifact can only affect the dead!</span>")
 		return
-
+	
+	for(var/mob/dead/observer/ghost in GLOB.player_list) //We put them back in their body
+		if(ghost.mind && ghost.mind.current == M && ghost.client)

```

Comment:
ghost.client is already true since you're pulling from GLOB.player_list

`GLOBAL_LIST_EMPTY(player_list)				//all mobs **with clients attached**.`